subsys/mpsl/mpsl_init.c

```diff
@@ -110,11 +110,21 @@ ISR_DIRECT_DECLARE(mpsl_radio_isr_wrapper)
 	return 1;
 }
 
+#if IS_ENABLED(CONFIG_MPSL_ASSERT_HANDLER)
+extern void mpsl_assert_handle(char *file, uint32_t line);
```

Comment:
If you define it with extern this way you don't get compilation errors if the application defines it wrong.
I would suggest either a function to register the handler, or defining the function in a header and the application will have to implement it.